packages/chip/src/Chip/Chip.styles.ts

```diff
@@ -39,175 +46,270 @@ export const chipWrapperPadding: Record<
+      },
+      [Property.Icon]: {
+        [InteractionState.Default]: palette.yellow.light2,
+      },
     },
     [Theme.Light]: {
-      color: palette.yellow.dark3,
-      bgColor: palette.yellow.light3,
-      focusBgColor: '#FFD664',
+      [Property.Background]: {
+        [InteractionState.Default]: palette.yellow.light3,
+        [InteractionState.Focus]: '#FFD664',
+      },
+      [Property.Text]: {
+        [InteractionState.Default]: palette.yellow.dark3,
+      },
+      [Property.Icon]: {
+        [InteractionState.Default]: palette.yellow.dark2,
+      },
     },
   },
+} as const satisfies RecursiveRecord<
+  [Variant, Theme, Property, InteractionState, string],
+  false
+>;
+
+const truncateDisabledColor = {
+  [Theme.Dark]: palette.gray.dark1,
+  [Theme.Light]: palette.gray.light1,
+};
+
+const wrapperDisabledColor = {
+  [Theme.Dark]: palette.gray.dark2,
+  [Theme.Light]: palette.gray.base,
 };
 
 /**
  * Chip wrapper
  */
-export const chipWrapperThemeStyle = (variant: Variant, theme: Theme) => css`
-  color: ${variantColor[variant][theme].color};
-  background-color: ${variantColor[variant][theme].bgColor};
-  transition: background-color ${transitionDuration.faster}ms ease-in-out;
-`;
-
-export const chipWrapperBaseStyle = css`
-  display: inline-flex;
-  align-items: center;
-  overflow: hidden;
-  white-space: nowrap;
-  box-sizing: border-box;
-  border-radius: 4px;
-`;
-
-export const chipWrapperSizeStyle = (baseFontSize: BaseFontSize) => css`
-  font-size: ${fontSize[baseFontSize]}px;
-  line-height: ${lineHeight[baseFontSize]}px;
-`;
-
-export const chipWrapperBaseDisabledStyles = css`
-  cursor: not-allowed;
-`;
-
-export const chipWrapperDisabledStyle: Record<Theme, string> = {
-  [Theme.Light]: css`
-    color: ${palette.gray.base};
-    background-color: ${palette.gray.light3};
-  `,
-  [Theme.Dark]: css`
-    color: ${palette.gray.dark2};
-    background-color: ${palette.gray.dark4};
-    box-shadow: inset 0 0 1px 1px ${palette.gray.dark2};
-  `,
-};
+export const getChipWrapperStyles = (
+  baseFontSize: BaseFontSize,
+  variant: Variant,
+  theme: Theme,
+  isDisabled = false,
+) =>
+  cx(
+    css`
+      display: inline-flex;
+      align-items: center;
+      overflow: hidden;
+      white-space: nowrap;
+      border-radius: 4px;
+      font-size: ${fontSize[baseFontSize]}px;
+      line-height: ${lineHeight[baseFontSize]}px;
+
+      color: ${variantColor[variant][theme].text.default};
+      background-color: ${variantColor[variant][theme].background.default};
+      transition: background-color ${transitionDuration.faster}ms ease-in-out;
+    `,
+    {
+      [css`
+        cursor: not-allowed;
+        background-color: ${color[theme].background.secondary.default};
+        color: ${wrapperDisabledColor[theme]};
+      `]: isDisabled,
+      [css`
+        box-shadow: inset 0 0 1px 1px ${palette.gray.dark2};
+      `]: isDisabled && theme === Theme.Dark,
```

Comment:
these are easier to parse when the css block is broken out into a human readable variable